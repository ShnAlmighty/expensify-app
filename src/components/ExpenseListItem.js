import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = (props) => (
    <div>
        <div>
            <Link to={`/edit/${props.id}`}><h3>{props.index}. {props.description}</h3></Link>
            <p>
            {numeral(props.amount).format('0,0.00')}Rs 
            : 
            {moment(props.createdAt).format('MMMM Do, YYYY')}
            </p>
        </div>
    </div>
);

export default ExpenseListItem;