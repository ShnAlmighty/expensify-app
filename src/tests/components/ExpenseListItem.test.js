import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem  from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render Expense List Item',()=>{
        const wrapper = shallow(<ExpenseListItem key={expenses[0].id} {...expenses[0]} index={expenses[0].id}/>);
        expect(wrapper).toMatchSnapshot();
});