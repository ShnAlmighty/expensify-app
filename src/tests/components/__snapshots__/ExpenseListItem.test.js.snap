// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense List Item 1`] = `
<div>
  <div>
    <Link
      to="/edit/1"
    >
      <h3>
        1
        . 
        gum
      </h3>
    </Link>
    <p>
      10.00
      Rs :
      January 1st, 1970
    </p>
  </div>
</div>
`;
